src/thunderbots/software/geom/ray.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <functional>
+
+#include "geom/line.h"
+#include "geom/point.h"
+#include "geom/segment.h"
+
+class Ray final
+{
+   public:
+    Point start;
```

Comment:
Really sorry about not noticing this in the earlier round, but can we have getters and setters for these too?

Reply:
Same with `Segment`, the comment did not show up for some reason

Reply:
No problem at all, TBH I probably should have thought of this on Saturday as well! Will update once I get home tonight.

Reply:
Just to make sure, getters and setters will be added in `ray`, `segment`, and already done so in `circle`. Would it be necessary to do so in `line` and `rectangle`?

Reply:
`Rectangle` is fine, it already has the necessary functions.
`Line` is a bit more contentious, I need to think about this a bit more. The difference between a `Line` and `Segment` is that a `Line` is infinite, and `Segment` is finite. The member variables currently exposed in `Line` don't feel right to me...

Reply:
Ok! 
`Segment` and `Ray` both have a `start` variable, so I decided to separate them into `get`/`setSegStart()` and `get`/`setRayStart()` accordingly.

Reply:
Thanks for pointing it out, sounds good. 
Added getters and setters for `Line`.